app/routers/downloads.py

```diff
@@ -78,6 +79,9 @@ async def download_application(
     :return: A streaming response with a zip file containing the documents.
     """
     with transaction_session(session):
+        dependencies.raise_if_unauthorized(
+            application, user, roles=(models.UserType.OCP, models.UserType.FI), scopes=(ApplicationScope.UNEXPIRED,)
+        )
```

Comment:
Maybe add support for scopes to `get_scoped_publication_as_user`. You can then change `get_publication_as_user` to `get_scoped_publication_as_user`, with the same roles and scopes arguments.

Reply:
Sounds good. Now that I think about it, should the get_*publication_* methods should be named get_*application_* instead?

Reply:
Uh, yes, haha, I must have been thinking about the registry subconsciously!